### PR TITLE
Lightbox component

### PR DESCRIPTION
**Pull request by NFSS10**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-components-react-native/issues/XXX |
| Decisions | Created `Lightbox` component  |
| Animated GIF | RN-Lightbox_1 <br>RN-lightbox_2 <br>RN-Lightbox_3 |

### CODE REVIEW

File: react/components/atoms/lightbox/lightbox.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        return (
+            <View>
+                <TouchableOpacity onPress={() => this.onLightboxPress()} activeOpacity={0.7}>
+                    <Image style={this._imageStyle()} source={{ uri: this.props.src }} />
+                </TouchableOpacity>
+                <Modal
+                    animationType="fade"
+                    transparent={false}
+                    visible={this.state.visible}
+                    onRequestClose={() => this.onBackButtonPress()}
```
Comment (gcandal) [inline react/components/atoms/lightbox/lightbox.js]: We're using `onStuff={this.props.something}` for functions, so let's go with that style.

File: react/components/atoms/lightbox/lightbox.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            },
+            this.props.style
+        ];
+    };
+
+    render() {
+        return (
+            <View>
+                <TouchableOpacity onPress={() => this.onLightboxPress()} activeOpacity={0.7}>
+                    <Image style={this._imageStyle()} source={{ uri: this.props.src }} />
```
Comment (gcandal) [inline react/components/atoms/lightbox/lightbox.js]: We need to potentially allow other kind of sources, so make:
- Rename `src` to `uri`
- Add a `source` prop
- Use `src` if defined, else use `source`

File: react/components/atoms/lightbox/lightbox.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    constructor(props) {
+        super(props);
+        this.state = {
+            visible: false
+        };
+    }
+
+    onBackButtonPress = () => {
+        this.setState({
+            visible: false
```
Comment (gcandal) [inline react/components/atoms/lightbox/lightbox.js]: Add a callback for visibility change

File: react/components/atoms/lightbox/lightbox.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        });
```
Comment (gcandal) [inline react/components/atoms/lightbox/lightbox.js]: Extract to a method

File: react/components/atoms/lightbox/lightbox.js (hunk 1 above)
Comment (BeeMargarida) [reply react/components/atoms/lightbox/lightbox.js]: Don't know if it makes sense, since this is a modal specific behaviour, contained in the component.

**Comment by gcandal** [comment]
@NFSS10 @BeeMargarida I did a few small changes:
- `Image`'s `source` to a computed (the 2 values were incoherent)
- Different style when passing functions, coherent with the rest of our components

**Pull request merged**